service/attack_surface.py

```diff
@@ -0,0 +1,95 @@
+import json
+import itertools
+
+
+class VirtualMachine:
+    def __init__(self, vm_id, name, tags):
+        self.vm_id = vm_id
+        self.name = name
+        self.tags = tags
+
+
+class FirewallRule:
+    def __init__(self, fw_id, source_tag, dest_tag):
+        self.fw_id = fw_id
+        self.source_tag = source_tag
+        self.dest_tag = dest_tag
+
+
+class AttackSurfaceService:
+    def __init__(self, input_file_path):
+        self.origin_service_input = self.get_service_input(input_file_path)
+        self.vms_data = self.parse_vms()
+        self.fw_rules_data = self.parse_fw_rules()
+        self.edges = self.prepare_data_graph()
+        self.request_count = 0
+        self.cumulative_process_time = 0
+        self.vms_number = len(self.vms_data)
+
+    @staticmethod
+    def get_service_input(input_file_path):
+        try:
+            with open(input_file_path) as json_file:
+                service_data = json.load(json_file)
```

Comment:
I don't think you need try and except when you are using 'with'. 
It does the work for you, read about it